### DISCUSSION TITLE
Implementing a transcription extension

### DISCUSSION THREAD

**Discussion by mr-islam** [standardnotes/forum#2481] [Suggestions]
Hi!
I just thought this could be a really great extension for Standard Notes. [oTranscribe](https://github.com/oTranscribe/oTranscribe) is a really great transcription service in itself, and I think it would not be too difficult to include it as an extension as well? 

Having it as a proper extension would save some major copy-and-pasting!

**Comment by atmoio** [comment]
Interesting..how would you imagine it working?

**Comment by mr-islam** [comment]
The oTranscribe "native" implementation (as in their webapp) opens a simple
text editor with support for bold and italics -- in Standard, that could
maybe be any of the available editor extensions... or maybe the
transcription would work better as its own editor?

Otherwise, it could be a component and sit at the bottom like the action
bar (the one that shows how many words written, etc.), with play, pause,
rewind, forward buttons (and all bound to keyboard shortcuts).

Selecting an audio file to transcribe might be messy I think (from my
limited programming knowledge) -- especially getting it to work on the
desktop apps and also the webapp.

**Comment by atmoio** [comment]
Well, I don't think this would be the type of extension I would work on officially, but my goal is to design the extension system so that oTranscribe (or anyone else for that matter) can easily build extensions like this if they want. The next few months should see some nice releases in this direction :)

**Comment by mr-islam** [comment]
I definitely understand, sounds good!

I'm dabbling in JS right now, and maybe in those few months I would be 
capable of doing something too :)
